integration-tests/emulated/src/tests/spiritnet/unpaid_execution.rs

```diff
@@ -0,0 +1,80 @@
+use crate::mock::{
+	network::MockNetworkPolkadot,
+	para_chains::{spiritnet, AssetHubPolkadot, AssetHubPolkadotPallet, Spiritnet},
+	relay_chains::Polkadot,
+};
+use frame_support::assert_ok;
+use parity_scale_codec::Encode;
+use xcm::{v3::WeightLimit, DoubleEncoded, VersionedMultiLocation, VersionedXcm};
+use xcm_emulator::{
+	assert_expected_events,
+	Instruction::{Transact, UnpaidExecution},
+	Junction, Junctions, OriginKind, Parachain, ParentThen, TestExt, Weight, Xcm,
+};
+
+#[test]
+fn test_unpaid_execution_to_spiritnet() {
+	MockNetworkPolkadot::reset();
+
+	let code = vec![];
+
+	let call: DoubleEncoded<()> = <Spiritnet as Parachain>::RuntimeCall::System(frame_system::Call::set_code { code })
+		.encode()
+		.into();
+	let sudo_origin = <AssetHubPolkadot as Parachain>::RuntimeOrigin::root();
+	let parachain_destination: VersionedMultiLocation =
+		ParentThen(Junctions::X1(Junction::Parachain(spiritnet::PARA_ID))).into();
+
+	let weight_limit = WeightLimit::Unlimited;
+	let require_weight_at_most = Weight::from_parts(1000000000, 200000);
+	let origin_kind = OriginKind::Superuser;
```

Comment:
Can we add the same also for the other `OriginKind` variants? It should be fairly easy to add.

Reply:
The unpaid execution should not care for the `OriginKind`. I simplified the tests: [4b5526b](https://github.com/KILTprotocol/kilt-node/pull/603/commits/4b5526b10cc8ee1aff1f4b2eba6417c5b6fd226b). 

I hope it is clearer now.